### PR TITLE
Move payment request notifications to Cloud Function

### LINKED ISSUE #428: Move payment request notifications to Cloud Function
Move the payment requests notification sending from the notification service to a database-triggered Cloud Function.

See https://www.notion.so/clabsco/Notification-Service-Indexer-064ca4f58be64c3384ac8fb5f00fca56 for more information on the motivation for doing this.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Moved the payment requests notification sending from the notification service to a database-triggered Cloud Function.

See https://www.notion.so/clabsco/Notification-Service-Indexer-064ca4f58be64c3384ac8fb5f00fca56 for more information on the motivation for doing this.

### Other changes

N/A

### Tested

Manually and added unit test. This is deployed in Alfajores.

### How others should test

Send a payment request to an address. See that the requested address receives a push notification.

### Related issues

- Fixes #428

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/cloud-functions/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+declare module '*.json' {
+  const value: any
+  export default value
+}
```
Comment (jeanregisser) [inline packages/cloud-functions/index.d.ts:4]: Can we avoid this? since you added `resolveJsonModule` to `tsconfig`

**Review by jeanregisser — COMMENTED** [review]
Looks great! 👍 
We need to add the new locale path to crowdin:
https://github.com/celo-org/wallet/blob/main/crowdin.yml

Also we'll need to think how to actually expose the metrics to prometheus. 
Since the cloud functions are short lived, prometheus won't be able to pull the metrics.

**Comment by gnardini** [comment]
I replaced the metrics for log-based ones. What we do here is just write logs as json and then we can convert the logs into metrics in the GCP console.

I created a dashboard where we can see [the number of sent notifications by type](https://console.cloud.google.com/monitoring/dashboards/builder/bca997df-6f7f-4b5e-8747-220b22b6e433?project=celo-mobile-mainnet&dashboardBuilderState=%257B%2522editModeEnabled%2522:false%257D&timeDomain=1h) (we only have `PAYMENT_REQUEST` as type for now). We can add more as time passes, I created a metric to track errors as well but since there's no data it's hard to create a visualization. When we get our first error I'll add that to the dashboard.

File: packages/cloud-functions/src/notifications/index.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        icon: 'ic_stat_rings',
+        color: '#42D689',
+      },
+    },
+    data,
+    token,
+  }
+
+  try {
+    // TODO: Use a logger that abstracts the json stringifying thing.
```
Comment (jeanregisser) [inline packages/cloud-functions/src/notifications/index.ts:67]: `firebase-functions` provides this out of the box: https://github.com/firebase/firebase-functions/issues/612#issuecomment-771762956

**Review by jeanregisser — APPROVED** [review]

File: packages/cloud-functions/src/notifications/index.ts (hunk 2 above)
Comment (gnardini) [reply packages/cloud-functions/src/notifications/index.ts:67]: I saw that library, but I wanted to use something that we could also use in the `blockchain-api` and the `notification-service` which is why I didn't use it. I will come back to this after I've made some progress on the cEUR stuff that I've left behind the last couple of weeks.

**Pull request merged**